### PR TITLE
refactor: migrate Tag to typescript

### PR DESCRIPTION
**Pull request by didoshotev**
## Description
Migrate [Tag](https://github.com/ethereum/ethereum-org-website/blob/dev/src/components/Tag.js) to typescript

## Related Issue

Refs: #6392

### CODE REVIEW

File: src/components/Tag.tsx [hunk 1]
```diff
@@ -46,21 +49,27 @@ const StyledIcon = styled(Icon)`
   margin-left: 1em;
 `
 
-const Tag = ({
+export interface IProps {
+  name: string
+  key: string
```
Comment (pettinarip) [inline src/components/Tag.tsx:54]: There is no need to specify `key` in this component. `key` is used on the consumer, when you need to do a `.map` or a loop but inside the component, you don't need to pass that as a prop.

**Review by pettinarip — CHANGES_REQUESTED** [review]
Looking great @didoshotev 💪🏼 thanks for the PR.

I've left a comment there to do some cleanup.

File: src/components/Tag.tsx (hunk 1 above)
Comment (didoshotev) [reply src/components/Tag.tsx]: Make sense, thank you.

**Review by pettinarip — APPROVED** [review]
Awesome @didoshotev 🚀

**Comment by pettinarip** [comment]
@didoshotev be sure to [join the discord](https://discord.gg/CetY6Y4) if you are interested in contributing further to the project or have any questions for the team. And we've just released our 2022 POAPs so remember to [claim yours](https://ethereum.org/en/contributing/#poap) also 🥳!

**Comment by pettinarip** [comment]
@all-contributors please add @didoshotev for code

**Pull request merged**